### PR TITLE
Cleanup verifyCredential/verifyPresentation arguments, pass objects [Jira: DCK-63]

### PR DESCRIPTION
**Pull request by SamHellawell**

### CODE REVIEW

File: src/utils/vc.js [hunk 1]
```diff
@@ -171,18 +164,11 @@ export async function verifyCredential(credential, resolver = null, compactProof
...
  * (signature by issuer) is correct.
- * @param {object} credential - verifiable credential to be verified.
- * @param {DIDResolver} resolver - Resolver for DIDs.
- * @param {Boolean} [compactProof] - Whether to compact the JSON-LD or not.
- * @param {Boolean} [forceRevocationCheck] - Whether to force revocation check or not.
- * Warning, setting forceRevocationCheck to false can allow false positives when verifying revocable credentials.
- * @param {object} [revocationAPI] - An object representing a map. "revocation type -> revocation API". The API is used to check
- * revocation status. For now, the object specifies the type as key and the value as the API, but the structure can change
- * as we support more APIs there are more details associated with each API. Only Dock is supported as of now.
+ * @param {object} [params] Verify parameters (TODO: add type info for this object)
```
Comment (lovesh) [inline src/utils/vc.js:167]: Same comment as above, credential is the primary argument and it should be outside the object.
Why can't the same comment bodies be kept for the params object?

File: src/utils/vc.js [hunk 2]
```diff
@@ -226,24 +212,14 @@ export async function signPresentation(presentation, keyDoc, challenge, domain,
...
- * @param {Boolean} [compactProof] - Whether to compact the JSON-LD or not.
- * @param {Boolean} [forceRevocationCheck] - Whether to force revocation check or not.
- * Warning, setting forceRevocationCheck to false can allow false positives when verifying revocable credentials.
- * @param {object} [revocationAPI] - An object representing a map. "revocation type -> revocation API". The API is used to check
- * revocation status. For now, the object specifies the type as key and the value as the API, but the structure can change
- * as we support more APIs there are more details associated with each API. Only Dock is supported as of now.
- * @param {object} [schemaAPI] - An object representing a map. "schema type -> schema API". The API is used to get a
- * schema doc. For now, the object specifies the type as key and the value as the API, but the structure can change
- * as we support more APIs there are more details associated with each API. Only Dock is supported as of now.
+ * @param {object} [params] Verify parameters (TODO: add type info for this object)
```
Comment (lovesh) [inline src/utils/vc.js:215]: Same comment as above

File: tests/integration/schema.test.js [hunk 3]
```diff
@@ -174,23 +178,37 @@ describe('Schema Blob Module Integration', () => {
...
     await expect(
-      verifyCredential(
-        invalidCredential.toJSON(), dockResolver, true, false, undefined, { notDock: dockApi },
-      ),
+        verifyCredential({
+          credential: invalidCredential.toJSON(),
+          resolver: null,
+          compactProof: true,
+          forceRevocationCheck: false,
+          revocationApi: undefined,
```
Comment (lovesh) [inline tests/integration/schema.test.js:186]: `revocationApi` can be omitted here and a few more places below

File: src/utils/vc.js [hunk 4]
```diff
@@ -132,21 +132,14 @@ export async function issueCredential(keyDoc, credential, compactProof = true) {
...
- * @param {Boolean} [compactProof] - Whether to compact the JSON-LD or not.
- * @param {Boolean} [forceRevocationCheck] - Whether to force revocation check or not.
- * Warning, setting forceRevocationCheck to false can allow false positives when verifying revocable credentials.
- * @param {object} [revocationAPI] - An object representing a map. "revocation type -> revocation API". The API is used to check
- * revocation status. For now, the object specifies the type as key and the value as the API, but the structure can change
- * as we support more APIs there are more details associated with each API. Only Dock is supported as of now.
- * @param {object} [schemaApi] - An object representing a map. "schema type -> schema API". The API is used to get
- * a schema doc. For now, the object specifies the type as key and the value as the API, but the structure can change
- * as we support more APIs there are more details associated with each API. Only Dock is supported as of now.
+ * @param {object} [params] Verify parameters (TODO: add type info for this object)
```
[the remark this answers is not in the document]
Comment (SamHellawell) [reply src/utils/vc.js:135]: yes, its not valid types for docs generation or tsc

[the remark this answers is not in the document]
Comment (SamHellawell) [reply src/utils/vc.js:135]: we can make a typeinfo for params object to show all the previous stuff to users, dont worry

File: src/utils/vc.js [hunk 5]
```diff
@@ -158,8 +151,8 @@ export async function verifyCredential(credential, resolver = null, compactProof
   });
 
   // Check for revocation only if the credential is verified and revocation check is needed.
-  if (credVer.verified && isRevocationCheckNeeded(credential.credentialStatus, forceRevocationCheck, revocationAPI)) {
-    const revResult = await checkRevocationStatus(credential, revocationAPI);
+  if (revocationApi && credVer.verified && isRevocationCheckNeeded(credential.credentialStatus, forceRevocationCheck, revocationApi)) {
```
[the remark this answers is not in the document]
Comment (SamHellawell) [reply src/utils/vc.js:154]: i was receiving an error about revocationApi being undefined calling from this method, will double check again though. maybe better to check if (!revocationApi) when needing verification and throwing an error is falsey

File: src/utils/vc.js (hunk 1 above)
Comment (SamHellawell) [reply src/utils/vc.js:167]: same reply as above

File: src/utils/vc.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (SamHellawell) [reply src/utils/vc.js:135]: but i do agree with credential and presentation being outside of the object

File: src/utils/vc.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (lovesh) [reply src/utils/vc.js:154]: Error means a crash or verification failure? You should get the latter if `forceRevocationCheck` is true and `revocationApi` is not supplied.

[the remark this answers is not in the document]
Comment (SamHellawell) [reply src/utils/vc.js]: disregard, was likely due to a naming issue as to why i got undefined error here (revocationAPI -> revocationApi during merging), no need for this change anymore so removed it

**Pull request merged**